### DISCUSSION TITLE
No way to disable auto-generated source code zip links

### DISCUSSION THREAD

**Discussion by kornelski** [community/community#6003] [Other Feature Feedback, Questions, & Ideas] [answered]
GitHub automatically adds "Source Code.zip" and "Source Code.tar.gz" links to every release. These are problematic for a few reasons:

1. When I make releases for non-technical audience (a binary build of my software), these links are unwanted. They add clutter and confuse non-developer users.

2. When my repo uses git submodules, these archives lack submodules' source code, which causes confusing compilation errors. I can't even _replace_ them with my own archives that contain complete source code.

3. The source tarballs are not even suitable for distribution of many projects in source code form, because they don't support inclusion of auto-generated files (e.g. `./configure` or compiled data files). Auto-generated files shouldn't be under version control — that causes repo bloat and merge pains.

So from my perspective the Source Code links are unnecessary, often undesirable, and even when I'd like some source code links, the auto-generated archives are incomplete and unfit for the purpose.

Please remove these links or at least add an option to disable them.

**Comment by MylesBorins** [comment, marked as answer]
Hey @kornelski 

This is a great idea and something we've talked about internally. We would need to do some research to see if there are any systems that rely on these artifacts that would break in fun and mysterious ways before we could follow through on this.

I've created a feature request internally. We are looking at a number of improvements that we could make to artifacts / assets, but we may not be funding this effort in the immediate future (to set expectations)

**Reply by bagder** [reply]
You could simply make it an option (perhaps per repo or even per release) to disable them. Then removing them wouldn't hurt users that need them.

**Reply by thunder-coding** [reply]
Perhaps, a better thing to do would be to some similar option to hide them from the GitHub UI instead of disabling them. (Helpful in certain scenarios and also to maintain a sort of backward compatibility for scripts configured to pick up the source tar/zip intentionally for some reasons)

**Reply by redtide** [reply]
> Perhaps, a better thing to do would be to some similar option to hide them from the GitHub UI instead of disabling them.

Why not disable something that is broken, or hide something that works? Sounds confusing and suspicious.

**Reply by lectrical** [reply]
It appears to partially resolve the issue, as a compromise, where other attempts to push for a complete solution, to what is a mixture of cosmetic and technical issues, are failing.

The cosmetic issue is the primary issue that should be resolved with a separate issue or discussion created to discuss what purpose they serve and what would be required to reform or remove them.

Resolving the cosmetic issue solves the core of the issue, so that no one sees or interacts with them, whilst avoiding any potential increase in work or support load that might arise from removing or changing them.

Hiding them from view is an elegant solution that will allow for a targeted discussion about their limitations and purpose, which will most likely never happen as we all just ignore them whilst breaking nothing. Much more likely to be implemented than a singular solution to all three points.

I don't think it's hard to see why they'd more likely do nothing than risk the alternative.

**Reply by penguin359** [reply]
Hello @MylesBorins Is there any update on the internal ticket request to get this feature live and active?

**Reply by Furimanejo** [reply]
I pray for this ticket every night before sleep, please implement a way to remove/hide these files 🙏

**Reply by flying-sheep** [reply]
Today's xz backdoor shows that allowing to replace those tarballs would be a horrible idea since it would allow malicious actors to sneak in code that doesn't exist in the commit history.

I'm pretty sure a lot of Linux distributions will make it policy to only use tarballs that directly come from tags for security reasons.

**Reply by penguin359** [reply]
> I'm pretty sure a lot of Linux distributions will make it policy to only use tarballs that directly come from tags for security reasons.

I really doubt that will be the case. First of all, there is already pre-existing technology to handle publishing secure tarballs: detached PGP signatures. There are many good projects that do provide these though many more that should be. This is the same cryptography technology that is used to create signed Git commits and tags so it has the same security model, but allows tarballs to be built and distributed securely regardless of what source control system is used to maintain the code by the developers.

Second, there are reasons why tarball contents don't match the repository contents one-for-one. There may be some files that are simply not worth re-distributing for end-users or package maintainers which may not need test datasets, or other development-only files. Also, more importantly, there may be a number of auto-generated files that should not be kept under source control but end-users are not expected to generate on their own such as automake/autoconf files or documentation generated from TeX or Doxygen. These would clutter up the source history, but end-users should not have to worry about installing extra software to generate them after a download.

**Reply by vladtcvs** [reply]
Source code tarballs shouldn't contain any data, which is not present in git repository. No autogenerated files

**Reply by flying-sheep** [reply]
What are “detached” PGP signatures?

FYI: the xz backdoor’s tarballs were signed by “Jia Tan”, the account that’s currently believed to be a malicious actor who gained the trust to maintain xz-utils for a while before introducing the backdoor.

**Reply by NaturellFil** [reply]
test

**Reply by foxheadworkshop** [reply]
> Hey @kornelski
> 
> This is a great idea and something we've talked about internally. We would need to do some research to see if there are any systems that rely on these artifacts that would break in fun and mysterious ways before we could follow through on this.
> 
> I've created a feature request internally. We are looking at a number of improvements that we could make to artifacts / assets, but we may not be funding this effort in the immediate future (to set expectations)

This has been a few years, can an updated be provided? An option to solve the usability issue while preserving the integration with dependencies would be to add a config on the release object to display/hide the default Source Code zip and tar files. 

This would have minimal impact and could be backwards compatible because it would only affect the UI not any APIs

**Reply by zyv** [reply]
Please just add a setting to disable these links. This is very problematic for projects that use build systems that require bootstrapping (such as autotools).

Right now, releases create links to the repository's exports without the bootstrapped build system completely confusing anyone looking at them.

**Reply by Relvl** [reply]
Bump. Please allow us to disable this unwanted archives.

**Reply by qpointsystems** [reply]
Bump. Amazing that this is considered a 'feature'.  It should actually default to NOT include source.  *heavy sigh*
Also, the zany 4 year delay here (even a "no, sorry, issue closed" would be fine) has me thinking github isn't the right place for my project long term. Good to know.

**Reply by gremlin529** [reply]
found this link when I was looking to do just this now, create releases with no source code in them. Please add another +1 to people who want this feature

**Reply by qpointsystems** [reply]
A conspiracy minded person would notice that GH is likely compensated by how much source code on their platform is legally consumable by the AI companies. For private projects, this mechanism we are discussing here as a side effect simply leads to more source being in safe-harbour for chewing by the various AI companies.  But that would be silly to think ;)

**Reply by Eurikon** [reply]
Unbelievable its 2026 and yet this is still not adressed… 
Please add an option inside the settings to disable this!

**Reply by mckayreedmoore** [reply]
I +1 for having this option. Seems odd that this wasn't just an "opt in" feature initially, but instead a default.

**Reply by joezappie** [reply]
This would be great. One of our devs just downloaded the source zip file and not our release zip file and burnt 10 minutes trying to help him figure out what was going on and while files would be missing. Would be more fool proof to just have our release files in the list

Or... at least have it formatted so built files are separate/hidden from the auto generated ones. 

This could probably be vibe coded in 3 minutes 😏

**Comment by bagder** [comment]
I'd like to add to this that in our project we upload our generated release tarballs to each release, but GitHub still adds the auto-generates ones at the bottom which then misleads some users pick those instead of the "real" release files listed just above, leading to problems explained above.

**Comment by redtide** [comment]
I opened a discussion years ago (I still keep [the link](https://github.community/t/disable-tarball/139966) but it's no longer valid, it was before discussions was added) about this issue but nothing since then (they also created an internal issue about it at that time, so who knows what happened).

Is it that difficult to add an option to disable, at organization/account level, other than per repository, the automatic creation of those mostly useless, confusing and annoying artifacts or is it the ancient Microsoft way of accommodating the majority of new/dumb users who need ready-made baby food? Are they so many that this was not managed/ignored for years?

Shortly after my and other requests I only saw the tar.gz format option disappear on the main page from the green "Code" button (formerly "Download"), which suffers from the same problem.

I heard other developers having to deal with their users about "broken packages" just because they use git submodules and automatic artifacts don't include them.

**Comment by redtide** [comment]
Suggestion:

Project repository => Settings => General => Features (not Archives, near LFS options could be confusing), something like:

- [x] Automatic package assets
  Creates tar.gz/zip package assets to download from the main page and when tagging new a release
---
If submodules are used by less users then this might be enabled by default unless an account/organization wide setting in:

Account/Org => Settings => Code, planning, and automation  => Repository => Repository defaults

same as repository' Features settings above

IMO this should be better disabled by default and add a tooltip somewhere to warn one time only new users that that option exists, otherwise a "Using submodules? You can disable this org or repo wide here and there" tooltip.

**Comment by xan2622** [comment]
Hi. 
Is it still being considered to add an option to disable these extra downloads (_Source Code.zip_ and _Source Code.tar.gz_) that are sometimes unwanted?

**Comment by dboeger1** [comment]
I hate to dredge up what appears to be an old, somewhat ignored topic, but I stumbled on this while having similar issues, and would really like to have the option to at least hide those artifacts. In DevOps/automation contexts, having control of source archive naming, publishing, and contents is crucial. While having these extra artifacts certainly doesn't block my ability to upload my own, it's just really annoying to see seemingly duplicate source archives with different names and have to explain to users that I have no clue if the GitHub one works with my code.

**Comment by ARogueOtaku** [comment]
Hey, I recently tried using the Git release feature for the 1st time and immediately noticed this. I see the original answer was back in Oct 2021 which was 2+ years ago. Were there any updates on this since then?

**Comment by redtide** [comment]
> We would need to do some research to see if there are any systems that rely on these artifacts that would break in fun and mysterious ways before we could follow through on this.

Is it that difficult to make this feature a repository settings option so it wouldn't cause problems to anyone else?

> but we may not be funding this effort in the immediate future (to set expectations)

[More than 3 years](http://web.archive.org/web/20211209104525/https://github.community/t/disable-tarball/139966) now.

**Comment by mcuee** [comment]
We have to put the same note for our releases.
https://github.com/libusb/libusb/releases/tag/v1.0.27
https://github.com/libusb/libusb/releases/tag/v1.0.26

Note:
*.tar.bz are our official release files, these are proper release tarballs including configure.

*.tar.gz and .zip are something GitHub generates, unfortunately we haven't found a way to disable them. They are just a git snapshot, without e.g. configure.

*) https://github.com/orgs/community/discussions/6003

**Comment by TheCarpetMerchant** [comment]
@MylesBorins Can you update us on this ? A simple option to disable the generation would be sufficient for basically everyone here.

**Comment by adamlui** [comment]
I just published my first release in a monorepo and immediately noticed how confusing the auto-attached files are.

@mcuee your solution is so elegant, I immediately [implemented it](https://github.com/adamlui/js-utils/releases/tag/minify.js-1.4.0) and it is beautiful workaround!

![image](https://github.com/community/community/assets/10906554/2a41a506-b0f4-4c80-a292-0dc2a90c389d)

**Reply by desaih** [reply]
what workaround you implemented? Adding a note?

**Reply by adamlui** [reply]
indeed, why does this anger you to the point of thumbs down i don't get it

**Reply by adamlui** [reply]
also how come you didn't thumb down the original innovator of this technique @mcuee when he shared it

**Reply by TheCarpetMerchant** [reply]
Because he's not presenting it as a solution or workaround, just highlighting that he had to do a dumb thing due to Github not giving us control over the release feature assets.

**Reply by adamlui** [reply]
wow now you too have reacted with the same angry thumbs down @TheCarpetMerchant . Did you know, workaround means work "around" the problem? i.e. he said what he had to do (write note) to work "around" it (contrary to your false claim it was not a workaround), and I confirmed this solution worked? So the only difference is, he presented it without the same eagerness as me, yet such a subtle difference elicited so much anger from you?

**Reply by redtide** [reply]
@adamlui OK, the more you go on with this ridiculous discussion I hope the more thumbs down you'll get, cause:
- you suppose an hypothetical anger from desaih when it was a clear act of ironic comment around your excitement for a simple note.
- the point was not about kidding for the workaround, but because "implementing that beautiful and elegant solution" just by copy a trivial note idea as workaround

this discussion is really about an issue that doesn't take a damn solution from years and we get people here happy by adding a note because it doesn't get solved, which makes it even more annoying, so please, stop this. Or are you a troll getting payed to make this closed?

**Reply by adamlui** [reply]
@redtide 

> OK, the more you go on with this ridiculous discussion I hope the more thumbs down you'll get

That is rather spiteful and no wonder my happy-toned comment is attracting all the angry people bitter I can have a pleasant tone, now I know how you angry guys think

> you suppose an hypothetical anger from desaih when it was a clear act of ironic comment around your excitement for a simple note

I supposed very real anger for one to go out of their way to thumbs down a helpful post made with positive  tone

> the point was not about kidding for the workaround, but because "implementing that beautiful and elegant solution" just by copy a trivial note idea as workaround

I don't know what this sentence means, can you re-word it please? (and indeed writing simple notes is a beauty to behold)

> this discussion is really about an issue that doesn't take a damn solution from years and we get people here happy by adding a note because it doesn't get solved, which makes it even more annoying, so please, stop this.

I know it doesn't get solved, GitHub doesn't care about many issues that linger for eternity, get used to it and be pleased when someone presents a beautiful solution (not sure how lessening the problem makes it "more annoying")

> Or are you a troll getting payed to make this closed?

You suggest GitHub financially contracted me to attract the angry people like you who use curse words at me to make this discussion "closed"??

**Reply by TheCarpetMerchant** [reply]
@adamlui This isn't a workaround. This is like having a button that provokes a crash and adding a note in the changelog that "BTW this buttons provokes a crash".

Calling it a workaround or a solution is an abuse of language, in France you'd get reported to the French academy for such things !

**Reply by adamlui** [reply]
@TheCarpetMerchant it is a workaround, it prevents the negative consequence from useless file attachment (reader's confusion) by telling reader, ignore the attachment (mitigating confusion from pointlessly analyzing attachment). With crash, negative consequence is death, so post-incident it cannot be workaround since death is irreversible, but pre-incident it is also workaround (discourages pressing death button mitigating consequence of death)

**Reply by adamlui** [reply]
Look how I even make it look like it might even be intended:

![image](https://github.com/community/community/assets/10906554/bc526939-5109-4ddb-acf5-b5b0e89a6b21)

See, you can turn lemon into lemonade by making readers think you're providing something extra of potential value

**Reply by TheCarpetMerchant** [reply]
You're absolutely hilarious, I'll be following for more !

**Reply by userdocs** [reply]
Listen, you people are thinking about this all wrong, instead of being critical of this clearly elegant and beautiful solution we should be doubling down here, expanding it's potential to be more eleganter. 

So I propose a spec of sorts, the most helpful note possible that can be created to aid the end user. I hope you understand the potential here.

The first suggestion I propose for this spec v0.0.1 is related to accessibility. Not everyone reads English, an ugly oversight if I do say so myself.

So I have translated the note into every possible language and suggest all notes to be included are translated to all known and unknown languages.

Here is my example: https://github.com/userdocs/test/releases/tag/v1.33.7

https://gist.github.com/userdocs/4e83e157d4cf57ca981b8e823dac821a

**Reply by adamlui** [reply]
@userdocs I got thousands of users in mere weeks which my internationalization attempts likely played at least a small part in, so your attempts to make me feel bad for turning a negative into a positive thus potentially captivating a wider audience is failing btw

**Reply by redtide** [reply]
you guys pay your attention to a troll.

**Reply by adamlui** [reply]
@redtide what troll do you speak of? 

![image](https://github.com/community/community/assets/10906554/2b6b09ed-b2ac-46af-b183-1b12c1fe17cc)

**Comment by mcuee** [comment]
> Because he's not presenting it as a solution or workaround, just highlighting that he had to do a dumb thing due to Github not giving us control over the release feature assets.

I do not understand the thumbdowns towards the comments by @adamlui. Maybe it is just due to the different understandings of the word `workaround`. 

For sure it is not `elegant` and it is not a `solution`. But we can not do much -- so we provide the note to let the user know which release tarball to download. You can call it sort of a `workaround` as well.

In the end, we all hope that github can provide the real solution so that we do not need to add the note in the future. For example, the suggestion to `make this feature a repository settings option` is a good solution to me.

**Reply by adamlui** [reply]
Solution may be pushing it, but it is very low-key elegant. It made my release notes seem richer and more professional thus probably responsible for the unusually high number of downloads I got for such a small and new project

**Reply by adamlui** [reply]
I wasn't joking it got like 3000 downloads after I added the note

**Reply by userdocs** [reply]
you know you can also do this right?  It's directly supported in GFM now. https://github.com/orgs/community/discussions/16925

![image](https://github.com/community/community/assets/16525024/98bca7ec-aa6d-4f79-ae96-c7cd0f99f6ef)

```
> [!NOTE]  
> Asset minify.js-1.4.0.tgz is the official release package [as published to npm](https://www.npmjs.com/package/@adamlui/minify.js/v/1.4.0). *.tar.gz and *.zip files are auto-generated by GitHub, they contain a git snapshot of the entire monorepo at the time of this release (which includes utilities besides [minify.js](https://github.com/adamlui/js-utils/tree/main/minify.js)).

> [!TIP]
> Asset minify.js-1.4.0.tgz is the official release package [as published to npm](https://www.npmjs.com/package/@adamlui/minify.js/v/1.4.0). *.tar.gz and *.zip files are auto-generated by GitHub, they contain a git snapshot of the entire monorepo at the time of this release (which includes utilities besides [minify.js](https://github.com/adamlui/js-utils/tree/main/minify.js)).

> [!IMPORTANT]  
> Asset minify.js-1.4.0.tgz is the official release package [as published to npm](https://www.npmjs.com/package/@adamlui/minify.js/v/1.4.0). *.tar.gz and *.zip files are auto-generated by GitHub, they contain a git snapshot of the entire monorepo at the time of this release (which includes utilities besides [minify.js](https://github.com/adamlui/js-utils/tree/main/minify.js)).

> [!WARNING]  
> Asset minify.js-1.4.0.tgz is the official release package [as published to npm](https://www.npmjs.com/package/@adamlui/minify.js/v/1.4.0). *.tar.gz and *.zip files are auto-generated by GitHub, they contain a git snapshot of the entire monorepo at the time of this release (which includes utilities besides [minify.js](https://github.com/adamlui/js-utils/tree/main/minify.js)).

> [!CAUTION]
> Asset minify.js-1.4.0.tgz is the official release package [as published to npm](https://www.npmjs.com/package/@adamlui/minify.js/v/1.4.0). *.tar.gz and *.zip files are auto-generated by GitHub, they contain a git snapshot of the entire monorepo at the time of this release (which includes utilities besides [minify.js](https://github.com/adamlui/js-utils/tree/main/minify.js)).
```

**Reply by adamlui** [reply]
Yes but the v-spacing is awkward and color brings unnecessary over-emphasis is why I avoid them

**Reply by adamlui** [reply]
GitHub designers added the coloring after I added notes/tips to my chatgpt.js.org and distinct coloring to sections/shields btw (but awkwardly added v-spacing so it doesn't look good like before).

They are one of my closest followers, frequently releasing "updates" after my innovations (another example, after I released my first ChatGPT Chrome extension, they copied my popup.html design for the re-design of upper-right corner button menu, with the shape and horizontal separators)

**Reply by adamlui** [reply]
Another example, after I added search to chatgpt.js.org, they overhauled their search engine and even redeisgned their website header row

**Reply by redtide** [reply]
> I do not understand the thumbdowns towards the comments by @adamlui. Maybe it is just due to the different understandings of the word workaround.

because attracts people like you wasting time and log talking about a ridiculous workaround while other people like me waits a solution from some years.
Then maybe some GH operator comes here like "let's see how is going that discussion... ah nice, they are focused on a stupid workaround, screw it."

**Reply by adamlui** [reply]
@redtide you can't even see, if a decision maker finds a workaround "stupid", let alone discussion is popping off like I make it, they will more likely address the problem

**Comment by rashed9810** [comment]
Can anyone please tell me how I want to be a  pro coder ?

**Reply by redtide** [reply]
you are doing good, by doing offtopic questions in the wrong place.

**Comment by peter-mccann** [comment]
I went down a rabbit hole, But I thought I'd share my findings anyway.

If you have made it this far congratulations. (please read everything)
If one is not opposed to using the github cli or api in a script here **WAS** a possible workaround.

[Github CLI release](https://cli.github.com/manual/gh_release)
[GitHub API release](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#delete-a-release-asset)

gh release delete-asset -y <tag> <asset-name> 

or using the API directly from [github discussions](https://github.com/cli/cli/discussions/3000)
The following returns JSON output
gh api repos/owner/repo/releases/tags/v1.0.0 
gh api -X DELETE <asset-url>

Note: This does require you to login to GH using a Personal Access Token as shown below
Note: Powershell does not have a way to redirect input into a command using "<" we are forced to use cmd.exe
        & cmd.exe /c "gh auth login --with-token < .\PAT.txt"
Note: From Unix
/usr/bin/gh auth login --with-token < ./PAT.txt

Verify login status
gh auth status

**Unfortunately**, The sourceCode.zip and sourceCode.tar.gz are not considered assets even though they are listed on the UI as Assets. They don't show up in the .assets[] json array when querying the releases
gh api repos/ORG/REPO/releases/tags/vX.Y.Z
  "assets": [],

**Comment by ckreibich** [comment]
It's been a couple of years and the original answer didn't offer any solutions, so I wanted to ask whether the planning for this has changed on Github's side? The pattern of "please use our tarballs, not the auto-generated ones" seems to have only become more pervasive since.

**Reply by babakfp** [reply]
Microsoft owns GitHub. So, don't expect anything! This is the same story. All of their projects slowly get fewer and fewer updates and fixes and after some point, they completely abandon them.

**Comment by zerefdev** [comment]
Bump

**Reply by vallerydelexy** [reply]
bump

**Comment by usmankhawar22** [comment]
We are getting around 100 Mbs of zip and tar file per releases, which is chocking our 10GB storage limit.

**Comment by MrGadget1024** [comment]
I need this too...just a setting in GitHub we can turn off so they're not published in the release.

![image](https://github.com/community/community/assets/9826063/2612c579-d34d-47e1-bdd5-a400d483832f)

**Comment by mloiseleur** [comment]
We need this too. 
Just a setting to be able to turn it off when it's not relevant.

**Comment by mnlx** [comment]
Bump!

**Comment by potus-barret** [comment]
```
     |     |     
  X  |  -  |  -  
 ____|_____|_____
     |     |     
  -  |  -  |  -  
 ____|_____|_____
     |     |     
  -  |  -  |  -  
     |     |     
```

**Comment by ghost** [comment]
Please allow this! For one of my repos, adding "source code" is misleading at best.

**Comment by LMarkowski** [comment]
Any update?

**Comment by helmavdl** [comment]
I wholeheartedly support the suggestion, in the meantime I found a little workaround that at least saves space:

- create a `.gitattributes` file in the root of your repo
- add the following line
```
* export-ignore 
```

This will create empty "source code" zips. Not the desired outcome, but at least it doesn't create 2 copies of the repo content.

**Reply by chrisrosner** [reply]
I did not find that this worked for me.

**Comment by GregTonoski** [comment]
I second the request.

**Comment by pustinia** [comment]
Please allow this !!!!!!!

**Comment by sht** [comment]
Guys, any update on this? 🤔

**Comment by ckreibich** [comment]
Yo Github, this would be _such a good holiday present_ to your community! And you have well over a week to do it still. What's not to like?

**Comment by lectrical** [comment]
All github has to let us do is hide it from view. That's it. It will break nothing and has 0 risk to them to just make it a cosmetic change. If you tell them to remove the actual assets you will probably break a silly amount of projects using the assets.  This is why they will never remove them but a cosmetic change to allow excluding these auto generated assets them from view in the website will break nothing.

Something like this you toggle in the repo settings, where if you have no custom assets, you see nothing.

Before

![2024-12-13_23-14-24](https://github.com/user-attachments/assets/f826da68-2922-4a70-89af-fa58464dfa5b)

After

![2024-12-13_23-11-23](https://github.com/user-attachments/assets/6478a7d8-ffaf-45f3-b52f-0bb3bf4718f3)

**Reply by foxheadworkshop** [reply]
I agree, some flag on the release config that lets me hide the default assets while displaying any custom ones added to the release.

**Comment by robtaylor** [comment]
Really, no movement on this for nearly 4.5 years? @MylesBorins ?

Is this possible on enterprise?

**Comment by RaiyanRizwan** [comment]
Need this! Working on a software where the source code is _not_ meant to be public alongside the executable release...

**Comment by peter-mccann** [comment]
Copilot AI returned this response when I requested a feature enhancement. (I submitted the ticket anyway) perhaps this will get some movement. 

Currently, GitHub does not provide a way to disable or delete the automatically generated source code archives (sourceCode.zip and sourceCode.tar.gz) that are included with releases. These archives are automatically created for every release based on the associated Git tag and are not considered part of the assets[] array when querying the release data via the GitHub API.

If you want to avoid these files being listed as part of your release, you could consider the following workarounds:

Custom Release Assets: Instead of relying on the auto-generated source code archives, you can manually upload your own release assets. This can be done through the GitHub UI or programmatically using the [Release Assets REST API](https://docs.github.com/en/enterprise-cloud@latest/rest/releases/assets).

Clarify in Release Notes: If the presence of these auto-generated files is causing confusion, you can add a note in your release description to clarify which files are the intended assets for your release.

Unfortunately, there is no built-in feature to disable or delete these auto-generated files at this time. If this is a critical need for your workflow, I recommend sharing your feedback directly with GitHub via the [GitHub Feedback form](https://support.github.com/contact/feedback?category=feedback). This ensures that your request is considered for future updates.

**Comment by lotsofthoughts** [comment]
I build and push my own binaries and it makes no sense why supplemental source code would be pushed along side them.

**Comment by ReedCopsey** [comment]
When using the releases as a company, providing "source code" to end users is often highly inappropriate. Please add a way to hide or remove these.

**Comment by robtaylor** [comment]
@tuves Any chance of someone picking this up?

**Comment by ssrszdx** [comment]
so slowly about the issue. I met the same question. please .....

**Comment by robtaylor** [comment]
I've re-raised it, as clearly no-one looks at older discussions... https://github.com/orgs/community/discussions/158761

**Comment by HungryHedgehog** [comment]
If I can't turn this off I can't use Git Hub to release betas for my game as it would contain undecrypted assets which would go against the licenses of said assets. There should be a way to only release the files we select without it autoincluding raw data.

**Comment by BuchaLizard** [comment]
./configure

**Comment by KittyBua** [comment]
Hey can me Help for arm64 Ubuntu? Cant Install libusb wine

**Comment by eyalroz** [comment]
(**edit:** My comment only regards repositories with actual source code. I have no opinion on non-source-code repositories.)

> 1. When I make releases for non-technical audience (a binary build of my software), these links are unwanted. They add clutter and confuse non-developer users.

This site holds git repositories. A release is, in particular, a release of that version of the repository. You can caution non-technical users to avoid the source tarballs.

> 2. When my repo uses git submodules, these archives lack submodules' source code, which causes confusing compilation errors. I can't even _replace_ them with my own archives that contain complete source code.

That's a bug with your repository. It can't "use git submodules". The repository must be usable independently of the VCS it uses (in your case, git). Fix your repo. Ways to do this:

1. Leverage a C++ package manager like Conan or vcpkg: With the relevant configuration file, the conan or vcpkg client will obtain the necessary dependencies.
2. Use your (meta-)build system to obtain the relevant dependencies directly, if they're missing (e.g. CMake's FetchContent mechanism); I wouldn't recommend this.
3. Give manual instructions regarding how to obtain the necessary dependencies

>  3. The source tarballs are not even suitable for distribution of many projects in source code form, because they don't support inclusion of auto-generated files (e.g. `./configure` or compiled data files). Auto-generated files shouldn't be under version control — that causes repo bloat and merge pains.

The tarballs contain a copy of your repository. Everything people need in order to build and use the code should be in there. If a file needs to be auto-generated on the build machine, then of course it shouldn't be in the repository; the person downloading it will generate it. With your example, the person downloading your repository will run the autoconf tool to obtain a `./configure` script.

> Please remove these links or at least add an option to disable them.

I would ask @MylesBorins and other GitHub people to specifically _decline_ doing this, even as option, because it encourages mis-management of repositories, and a mis-perception of a repository maintainer's responsibility when making releases.

**Reply by ReedCopsey** [reply]
> This site holds git repositories. A release is, in particular, a release of that version of the repository. You can caution non-technical users to avoid the source tarballs.

The fact that GitHub specifically recommends things such as [Issue Only Repositories](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-an-issues-only-repository) shows that the site is for more than just holding "git repositories". That may be the primary purpose, but there are many use cases where it makes sense to not force these to exist.

**Reply by eyalroz** [reply]
@ReedCopsey : Ok, fair enough, but - such repositories would probably not have releases, right?

**Reply by ReedCopsey** [reply]
No - releases are the recommend way to close issues and tie them to a release (managed separately for code) in those repositories. That's pretty much my exact pain point at the moment. The releases make sense for managing issues, but the stupid "release artifact" files are pretty awful for end users looking at them.

**Reply by eyalroz** [reply]
I see. Well, I've qualified my comment to only refer to repositories with actual source code.

**Reply by gian-didom-emovo** [reply]
I didn't know using `git submodule` meant having a bug in my repository. You never stop learning.

**Comment by foxheadworkshop** [comment]
My actions only run when sings parts of my repo are updated. Those parts create two sets of zips (software and STLs)

The rest of the repo is guidance, and internal files that I don't really want users downloading. 

It really is a simple request. Allow for the release to be configured to show those links or not. 

I can add custom files,  zips etc. Why can't I hide the default?

**Comment by Abdumajidov2005** [comment]
Submodules & generated files

If your project uses submodules, include them in your custom archive.

You can also include generated files (like ./configure outputs or compiled data) that you don’t store in Git.

Inform your users

Add a short note in the release description: “Ignore GitHub’s Source Code.zip — use the provided custom archive instead.”

**Comment by lbmeng** [comment]
Can't believe this issue is still not addressed as of April 2026 ...
